plugin/trino-kafka-event-listener/src/main/java/io/trino/plugin/eventlistener/kafka/KafkaEventListenerConfig.java

```diff
@@ -0,0 +1,217 @@
+    {
+        return clientId;
+    }
+
+    @Config("kafka-event-listener.client-id")
+    public KafkaEventListenerConfig setClientId(String clientId)
+    {
+        this.clientId = Optional.ofNullable(clientId);
+        return this;
+    }
+
+    public Optional<String> getCompletedTopicName()
+    {
+        return completedTopicName;
+    }
+
+    @Config("kafka-event-listener.completed-event.topic")
+    public KafkaEventListenerConfig setCompletedTopicName(String completedTopicName)
+    {
+        this.completedTopicName = Optional.ofNullable(completedTopicName);
+        return this;
+    }
+
+    public Optional<String> getCreatedTopicName()
+    {
+        return createdTopicName;
+    }
+
+    @Config("kafka-event-listener.created-event.topic")
+    public KafkaEventListenerConfig setCreatedTopicName(String createdTopicName)
+    {
+        this.createdTopicName = Optional.ofNullable(createdTopicName);
+        return this;
+    }
+
+    public boolean getPublishCreatedEvent()
+    {
+        return publishCreatedEvent;
+    }
+
+    @ConfigDescription("Whether to publish io.trino.spi.eventlistener.QueryCreatedEvent")
+    @Config("kafka-event-listener.publish-created-event")
+    public KafkaEventListenerConfig setPublishCreatedEvent(boolean publishCreatedEvent)
+    {
+        this.publishCreatedEvent = publishCreatedEvent;
+        return this;
+    }
+
+    public boolean getPublishCompletedEvent()
+    {
+        return publishCompletedEvent;
+    }
+
+    @ConfigDescription("Whether to publish io.trino.spi.eventlistener.QueryCompletedEvent")
+    @Config("kafka-event-listener.publish-completed-event")
+    public KafkaEventListenerConfig setPublishCompletedEvent(boolean publishCompletedEvent)
+    {
+        this.publishCompletedEvent = publishCompletedEvent;
+        return this;
+    }
+
+    public Set<String> getExcludedFields()
+    {
+        return this.excludedFields;
+    }
+
+    @ConfigDescription("Comma-separated list of field names to be excluded from the Kafka event (their value will be replaced with null). E.g.: 'payload,user'")
+    @Config("kafka-event-listener.excluded-fields")
+    public KafkaEventListenerConfig setExcludedFields(Set<String> excludedFields)
+    {
+        this.excludedFields = excludedFields.stream()
+                .filter(field -> !field.isBlank())
+                .collect(toImmutableSet());
+        return this;
+    }
+
+    public Map<String, String> getKafkaClientOverrides()
+    {
+        return this.kafkaClientOverrides;
+    }
+
+    @ConfigDescription("Comma-separated list of key-value pairs to specify kafka client config overrides. E.g.: 'buffer.memory=67108864,compression.type=zstd'")
+    @Config("kafka-event-listener.client-config-overrides")
+    public KafkaEventListenerConfig setKafkaClientOverrides(String kafkaClientOverrides)
+    {
+        this.kafkaClientOverrides = kafkaClientOverrides == null ? Collections.emptyMap() : MAP_SPLITTER.split(kafkaClientOverrides);
```

Comment:
just disallow nulls: `MAP_SPLITTER.split(requireNonNull(kafkaClientOverrides, "kafkaClientOverrides is null"));`